src/udriveslist.pas

```diff
@@ -591,7 +591,8 @@ procedure TDrivesListPopup.UpdateCells;
           Format('%s/%s', [cnvFormatFileSize(FreeSize, uoscHeader),
                            cnvFormatFileSize(TotalSize, uoscHeader)])
       end
-      else if (Drive^.DriveSize > 0) then
+      else if IsAvailable(Drive, False) and
```

Comment:
Bad code, it will break a disk size for unmounted normal disks: HDD, Flash etc.
If `DeviceSize` is wrong for CD disks then simply zero it in the `uUDev` for CD disks.

Reply:
Please see the new fix